src/Microsoft.Net.Http.Headers/HttpRuleParser.cs

```diff
@@ -241,6 +241,11 @@ internal static string DateToString(DateTimeOffset dateTime)
 
         internal static bool TryStringToDate(string input, out DateTimeOffset result)
         {
+            if (input == null)
```

Comment:
Does this help to add a check for empty strings or is that not a common occurence?